ush/python/pygfs/task/stage_ic.py

```diff
@@ -1,58 +1,470 @@
+
+        Raises
+        ------
+        ValueError
+            If RUN type is unknown or GEFSTYPE is invalid for GEFS runs
+        """
+
+        if stage_dict.RUN == 'gefs':
+            gefstype = self.task_config.get('GEFSTYPE', None)
+            if gefstype == 'gefs-real-time':
+                return self._paths_from_templates(stage_dict, self._get_member_com_paths_gefs_rt(stage_dict, member))
+            elif gefstype == 'gefs-offline':
+                return self._paths_from_templates(stage_dict, self._get_member_com_paths_gefs_offline(stage_dict, member))
+            else:
+                raise ValueError(f"Invalid GEFSTYPE '{gefstype}' for RUN 'gefs'.")
+        elif stage_dict.RUN in ('gcafs', 'enkfgdas', 'gcdas', 'gdas'):
+            return self._paths_from_templates(stage_dict, self._get_member_com_paths_gcafs(stage_dict, member))
+        elif stage_dict.RUN == 'gfs':
+            return self._paths_from_templates(stage_dict, self._get_member_com_paths_gfs(stage_dict, member))
+        else:
+            raise ValueError(f"Unknown RUN type: {stage_dict.RUN}")
+
+    @staticmethod
+    def get_member_list(run: str, nmem_ens: int, m_index: int = 0, gefstype: Optional[str] = None) -> list[int]:
+        """Get list of member indices based on RUN type
+
+        Parameters
+        ----------
+        run : str
+            RUN type (e.g., 'gfs', 'gefs', 'enkfgdas', 'gdas', 'gcafs')
+        nmem_ens : int
+            Total number of ensemble members
+        m_index : int, optional
+            Cycle index for GEFS real-time mapping (default: 0)
+        gefstype : Optional[str], optional
+            GEFS type ('gefs-real-time' or 'gefs-offline')
+
+        Returns
+        -------
+        list[int]
+            List of member indices to process
+        """
+        if run in ['enkfgdas']:
+            return list(range(1, nmem_ens + 1))
+        elif run in ['gefs']:
+            if gefstype == 'gefs-real-time':
+                # Map GEFS members to GFS member numbers
+                # Cycle ranges determine which 30-member range to use based on cycle index
+                cyc_ranges = [
+                    list(range(1, 31)),
+                    list(range(21, 51)),
+                    list(range(41, 71)),
+                    list(range(61, 81)) + list(range(1, 11))
+                ]
+                member_list = [0]  # Start with control member
+                for gefs_member in range(1, nmem_ens + 1):
+                    gfs_member = cyc_ranges[m_index][gefs_member - 1]
+                    member_list.append(gfs_member)
+                return member_list
+            else:
+                # GEFS offline uses sequential member numbering
+                return list(range(0, nmem_ens + 1))
+        else:
+            return []
+
+    def _paths_from_templates(self, stage_dict: AttrDict, com_path_tuples: Tuple[Tuple[str, str, Dict[str, Any]], ...]) -> Dict[str, str]:
```

Comment:
```suggestion
    @staticmethod
    def _paths_from_templates(stage_dict: AttrDict, com_path_tuples: Tuple[Tuple[str, str, Dict[str, Any]], ...]) -> Dict[str, str]:
```
This method no longer needs `self``